### PR TITLE
AFE: POST to AFE's Pardot instance

### PR DESCRIPTION
**Pull request by islemaster**
Gives our AFE controller the ability to post to AFE's Pardot form handler endpoint in the correct format.

The endpoint URL is configurable, and stored in Secrets Manager for the production environment.  You should add it to locals.yml if you're doing active development on this feature.

Covered by unit tests.  Not used in production yet.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by bencodeorg — APPROVED** [review]

**Comment by bencodeorg** [comment]
Note I've uploaded the secret in this PR.

File: dashboard/test/controllers/api/v1/amazon_future_engineer_controller_test.rb [hunk 1]
```diff
@@ -2,46 +2,94 @@
...
-        'inspirational-marketing-kit' => '0',
-        'csta-plus' => '0',
-        'aws-educate' => '0',
-        'amazon-terms' => '1',
-        'new-code-account' => '0',
-        'registration-date-time' => '1997-07-16T19:20:30+01:00'
-      }, as: :json
-
-    assert_response :success, "Failed response: #{response.body}"
+    assert_response :bad_request, "Expected BAD REQUEST, got: #{response.status}\n#{response.body}"
```
Comment (bencodeorg) [inline dashboard/test/controllers/api/v1/amazon_future_engineer_controller_test.rb:34]: One note is we might actually want `response.code` (assuming the `FakeResponse` is right)?

Comment (bencodeorg) [reply dashboard/test/controllers/api/v1/amazon_future_engineer_controller_test.rb:34]: NVM, tested and `status` works as well. A little confused by [docs here](https://api.rubyonrails.org/classes/ActionDispatch/Response.html#method-i-code), does `status` represent a way to set the status of a response, or get it, or both?

Comment (islemaster) [reply dashboard/test/controllers/api/v1/amazon_future_engineer_controller_test.rb:34]: Looking at those docs: `status` is both a readonly attribute on a response object, documented here:

image

...and also the name of a method that can be used to set the `status` attribute, documented here:

image

Finally, it looks like the `code` method is only different from `status` in that it always returns a string, where status codes are normally numbers.

image

**Pull request merged**